llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -21048,6 +21048,11 @@ bool RISCVTargetLowering::isLegalInterleavedAccessType(
       return false;
 
     ContainerVT = getContainerForFixedLengthVector(VT.getSimpleVT());
+  } else {
+    // The intrinsics for scalable vectors are not overloaded on pointer type
+    // and can only handle the default address space.
+    if (AddrSpace)
+      return false;
```

Comment:
Nit, you could remove the else here

Reply:
The previous `if` block doesn't unconditionally return, so taking out the `else` would alter the behaviour and also apply the check to fixed-length vector types, wouldn't it? Or am I misreading?

Reply:
Sorry you're right, I misread it